### ISSUE TITLE
How to Deploy in Production

### ISSUE THREAD

**Issue by BenMiriello** [twilio/twilio-video-app-react#410]
First, I'm using my own authentication server, not the client-side server.js/passcode system this demo version comes with. I have the site (and my separate token server) deployed on Heroku rather than using the twilio-serverless deploy method or hosting.

Admittedly, this may as much a React question than a twilio-video-app-react question, but I can't find any good direction on how to deploy this in production mode, so I'm asking here. Could you recommend what package.json scripts or other configuration I'd need to change in order to run in production?

Thanks!

**Comment by timmydoza** [comment]
Hey @BenMiriello - thanks for the question!

This app is built with create react app, so all of their documentation applies to this app: https://create-react-app.dev

It looks like the `npm run build` script will build production-ready assets in the `build/` folder. More info [here](https://create-react-app.dev/docs/production-build).

Additionally, if you'd like to customize your production build, there are some options that are available by setting environment variables: https://create-react-app.dev/docs/advanced-configuration

Once the production assets are built, they will have to be served somehow. There are many recommendations on [this page](https://create-react-app.dev/docs/deployment). Looks like they have a [recommendation for Heroku](https://create-react-app.dev/docs/deployment#heroku), too.

Please let me know if you have any other questions! We're happy to help!

**Comment by BenMiriello** [comment]
Thanks for taking the time to respond and giving some direction in making this work. Much appreciated.

**Comment by BenMiriello** [comment]
For the sake of anyone else trying to do the same thing, I'll share the method that worked for me:

- set the start script to `npm run build` as mentioned above  
- create a [Procfile](https://devcenter.heroku.com/articles/procfile) and in it write: `web: node server.js`  

This isn't entirely necessary, but since I wasn't using [server.js](server.js) as a token server anymore, I simplified it to this:
```javascript
const express = require('express');
const app = express();
const path = require('path');
const PORT = process.env.PORT || 8080;
const HOST = process.env.HOST || '0.0.0.0';

app.use(express.static(path.join(__dirname, 'build')));

app.get('*', (_, res) => res.sendFile(path.join(__dirname, 'build/index.html')));

app.listen(PORT, HOST, () => console.log('server running on %d', PORT));
```

Besides the links shared by @timmydoza, [this walkthrough by Jason Cote](https://jasoncote.co/deploy-production-build-create-react-app-to-heroku) gave me the most guidance. Just note that the Procfile example in that article has a syntax error. Instead use the line I shared here.
